### DISCUSSION TITLE
Is it possible to messure e-commerce events with the analytics plugin?

### DISCUSSION THREAD

**Discussion by mariusbolik** [capawesome-team/capacitor-firebase#330] [Q&A]
Hello,

thank you for the great plugin!
Is it possible to measure e-commerce events exactly like mentioned here: https://firebase.google.com/docs/analytics/measure-ecommerce
Should I simply use logEvent method or is there any other preferred approach?

Regards,
Marius

**Comment by robingenz** [comment]
Hi,
yes, this should be possible using the `logEvent` method.

The correct event names can be found on [this page](https://firebase.google.com/docs/analytics/measure-ecommerce#web-version-9_1) if you select `Web version 9` as the SDK language.  
Example:
```ts
logEvent(analytics, 'view_item_list', params1);
```

It may be that there are problems with the nested objects:
```ts
// A pair of jeggings
const item_jeggings = {
  item_id: 'SKU_123',
  item_name: 'jeggings',
  item_category: 'pants',
  item_variant: 'black',
  item_brand: 'Google',
  price: 9.99
};

const params1 = {
  item_list_id: 'L001',
  item_list_name: 'Related products',
  items: [item_jeggings, item_boots, item_socks]
};
```
But this would be a bug that can be fixed. So feel free to let me know if it works.

**Reply by mariusbolik** [reply]
Thank you @robingenz, I'll try! 😃
